### PR TITLE
[MAINT] Clear measurement info when stopping hpi plugin

### PR DESCRIPTION
**Pull request by gabrielbmotta**
- Cleared saved Fiff Info and Digitizer Data when resetting plugin state.

### CODE REVIEW

**Review by juangpc — APPROVED** [review]

**Review by LorenzE — DISMISSED** [review]

File: applications/mne_scan/plugins/hpi/hpi.cpp [hunk 1]
```diff
@@ -153,6 +153,7 @@ bool Hpi::start()
 
 bool Hpi::stop()
 {
+    requestInterruption();
```
Comment (LorenzE) [inline applications/mne_scan/plugins/hpi/hpi.cpp:156]: IF you ask for a request interruption here, you would need to include a check for interruption in the run() method.

File: applications/mne_scan/plugins/hpi/hpi.cpp [hunk 2]
```diff
@@ -163,6 +164,9 @@ void Hpi::resetState()
 {
     m_bPluginControlWidgetsInit = false;
     m_pCircularBuffer->clear();
+
+    m_pFiffInfo.clear();
```
Comment (LorenzE) [inline applications/mne_scan/plugins/hpi/hpi.cpp:168]: When is the resetState() called. Please be aware that all plugins share one single fiff info. So clearing it here will also clear it for all other plugins.

Comment (juangpc) [reply applications/mne_scan/plugins/hpi/hpi.cpp:168]: Wow.

Comment (gabrielbmotta) [reply applications/mne_scan/plugins/hpi/hpi.cpp:168]: I am clearing the pointer instance, which just drops the reference by one.

File: applications/mne_scan/plugins/hpi/hpi.cpp (hunk 1 above)
Comment (gabrielbmotta) [reply applications/mne_scan/plugins/hpi/hpi.cpp:156]: The loop in the run method of the hpi fit is `while(!isInterruptionRequested())`. Do you mean add more break points with that condition inside the loop?

File: applications/mne_scan/plugins/hpi/hpi.cpp (hunk 2 above)
Comment (LorenzE) [reply applications/mne_scan/plugins/hpi/hpi.cpp:168]: @gabrielbmotta ah ok. Now I see it too. That should be fine then :)
@juangpc Comments like "wow." are not very expressive. What would you like to say here?

File: applications/mne_scan/plugins/hpi/hpi.cpp (hunk 1 above)
Comment (LorenzE) [reply applications/mne_scan/plugins/hpi/hpi.cpp:156]: sry my bad. isInterruptionRequested() is used already indeed.

File: applications/mne_scan/plugins/hpi/hpi.cpp (hunk 2 above)
Comment (juangpc) [reply applications/mne_scan/plugins/hpi/hpi.cpp:168]: Yes. Sry for the noise. Deleted.

**Review by LorenzE — APPROVED** [review]

**Review by juangpc — APPROVED** [review]

**Pull request merged**